.github/workflows/main.yaml

```diff
@@ -41,14 +41,23 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Create branch from commit by event name
+        run: |
+          if [[ '${{ github.event_name }}' == 'push' ]]; then
+            git branch __branch_before ${{ github.event.before }}
+          elif [[ "${{ github.event_name }}" == 'pull_request' ]]; then
+            git branch __branch_before ${{ github.event.pull_request.base.sha }}
+          fi
       - name: Set up JDK ${{ matrix.java_version }}
         uses: actions/setup-java@v4
         with:
           java-version: ${{ matrix.java_version }}
           distribution: 'temurin'
           cache: 'maven'
       - name: Build with JDK ${{ matrix.java_version }}
-        run: mvn --batch-mode install -DskipITs ${{ matrix.included_modules }}
+        run: mvn --batch-mode install ${{ matrix.included_modules }} -Dgib.disable=false -Dgib.referenceBranch=__branch_before
```

Comment:
hi @famod, i remove the gib.buildDownStream and gib.buildUpstream now
no more different compare to before

Reply:
Ok, makes more sense now.

One crucial thing to beware of is that full (or almost full) builds will take much longer now than before.
This can happen when e.g. `langchain4j-parent/pom.xml` is changed.
Are timeouts, memory sizings etc. ready for such a worst case scenario?

Reply:
> One crucial thing to beware of is that full (or almost full) builds will take much longer now than before. This can happen when e.g. `langchain4j-parent/pom.xml` is changed. Are timeouts, memory sizings etc. ready for such a worst case scenario?

@langchain4j i can't run all moudle build with IT enabled,may be you could help to check this scenario?

Reply:
@Yellow-- to check this, could you please do some change in `langchain4j-parent/pom.xml`? I can trigger the build then if you won't be able to do so

Reply:
hi @langchain4j , i have add some safe changed to ```langchain4j-parent/pom.xml```

Reply:
@Yellow-- thanks! I've started the builds